src/events/messageCreate.ts

```diff
@@ -200,6 +211,82 @@ const convertResumePdfsIntoImages = async (
   }
 };
 
+const countingGameLogic = async (
+  client: Client,
+  message: Message,
+): Promise<Message<boolean> | undefined> => {
+  // Check to see if game should end
+  let reasonForFailure = '';
+  if (isNaN(Number(message.content))) {
+    // Message was not a number
+    reasonForFailure = `"${message.content}" is not a number!`;
+  } else if (previousCountingAuthors.find((author) => author === message.author)) {
+    // Author is still on cooldown
+    reasonForFailure = `<@${message.author.id}> counted too recently!`;
+  } else if (Number(message.content) != currentCountingNumber) {
+    // Wrong number was sent
+    reasonForFailure = `${message.content} is not the next number! The next number was ${currentCountingNumber}.`;
+  }
+
+  if (reasonForFailure) {
+    return endCountingGame(client, message, reasonForFailure);
```

Comment:
Subtract currentCountingNumber by 1 when the game ends because this value of currentCountingNumber was not actually reached.